kola/tests/ignition/mount.go

```diff
@@ -0,0 +1,420 @@
+						}
+					},
+					{
+						"path": "/var/log/hello.txt",
+						"contents": {
+							"source": "data:,hello%20world%0A"
+						}
+					}
+				]
+			},
+			"systemd": {
+				"units": [
+					{
+						"name": "var-lib-containers.mount",
+						"enabled": true,
+						"contents": "[Mount]\nWhat=/dev/disk/by-partlabel/CONTR\nWhere=/var/lib/containers\nType=xfs\nOptions=defaults\n[Install]\nWantedBy=local-fs.target"
+					},
+					{
+						"name": "var-log.mount",
+						"enabled": true,
+						"contents": "[Mount]\nWhat=/dev/disk/by-partlabel/LOG\nWhere=/var/log\nType=xfs\nOptions=defaults\n[Install]\nWantedBy=local-fs.target"
+					}]
+			}
+		}`),
+		ClusterSize: 1,
+		Platforms:   []string{"qemu"},
+		Distros:     []string{"fcos", "rhcos"},
+	})
+}
+
+// Mount two disks with id `virtio-secondary-disk` and `virtio-tertiary-disk`
+// and make sure we can write files to the mount points
+func testMountDisks(c cluster.TestCluster) {
+	var m platform.Machine
+	var err error
+	var ignConfig *conf.UserData
+
+	options := platform.MachineOptions{
+		AdditionalDisks: []platform.Disk{
+			{Size: "1024M", DeviceOpts: []string{"serial=secondary-disk"}},
+			{Size: "1024M", DeviceOpts: []string{"serial=tertiary-disk"}},
+		},
+	}
+
+	// TODO: use translation between Ignition v2 and v3 in the future
+	switch c.IgnitionVersion() {
+	case "v2":
+		ignConfig = v2IgnitionConfig
+	case "v3":
+		ignConfig = v3IgnitionConfig
+	default:
+		c.Fatal("unsupported ignition version")
+	}
+
+	switch pc := c.Cluster.(type) {
+	// These cases have to be separated because when put together to the same case statement
+	// the golang compiler no longer checks that the individual types in the case have the
+	// NewMachineWithOptions function, but rather whether platform.Cluster does which fails
+	case *qemu.Cluster:
+		m, err = pc.NewMachineWithOptions(ignConfig, options, true)
+	case *unprivqemu.Cluster:
+		m, err = pc.NewMachineWithOptions(ignConfig, options, true)
+	default:
+		c.Fatal("unknown cluster type")
+	}
+
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// reboot it to make sure it comes up again
+	err = m.Reboot()
+	if err != nil {
+		c.Fatalf("could not reboot machine: %v", err)
+	}
+
+	// make sure partitions are mounted and files are written
+	mountContents := c.MustSSH(m, "cat /proc/mounts")
+	mountValidate(c, m, string(mountContents), "/var/lib/containers")
+	mountValidate(c, m, string(mountContents), "/var/log")
+}
+
+func testMountPartitions(c cluster.TestCluster) {
+	m := c.Machines()[0]
+	// make sure partitions are mounted and files are written
+	mountContents := c.MustSSH(m, "cat /proc/mounts")
+	mountValidate(c, m, string(mountContents), "/var/lib/containers")
+	mountValidate(c, m, string(mountContents), "/var/log")
+}
+
+// Validate partition is mounted to `path` and `path`/hello.txt is written
+func mountValidate(c cluster.TestCluster, m platform.Machine, mountContents, path string) {
+	if !strings.Contains(mountContents, path) {
```

Comment:
I'd add a space to the `path` here so that e.g. `/var/log/systemd` doesn't match.

Reply:
SGTM! Updated.